clap_mangen/src/lib.rs

```diff
@@ -244,8 +244,38 @@ impl Man {
     }
 
     fn _render_options_section(&self, roff: &mut Roff) {
-        roff.control("SH", ["OPTIONS"]);
-        render::options(roff, &self.cmd);
+        let help_headings = self
+            .cmd
+            .get_arguments()
+            .filter_map(|arg| arg.get_help_heading())
+            .fold(vec![], |mut acc, header| {
+                if !acc.contains(&header) {
+                    acc.push(header);
+                }
+
+                acc
+            });
+
+        let (args, mut args_with_heading) =
+            self.cmd
+                .get_arguments()
+                .filter(|a| !a.is_hide_set())
+                .partition::<Vec<_>, _>(|a| a.get_help_heading().is_none());
+
+        if !args.is_empty() {
+            roff.control("SH", ["OPTIONS"]);
+            render::options(roff, &args);
+        }
+
+        for heading in help_headings {
+            let args;
+            (args, args_with_heading) = args_with_heading
+                .into_iter()
+                .partition(|&a| a.get_help_heading() == Some(heading));
+
+            roff.control("SH", [heading.to_uppercase().as_str()]);
```

Comment:
If a help heading is used on a positional, then this will print out headings without anything underneath them

Reply:
I don't see how this is true, positionals should be treated the same. 

I did notice that this was an issue for hidden arguments so fixed that.

Reply:
We get the headers for all `Arg`s and then `option`s filters out positionals

Reply:
`reader::option` is used for both. It first prints non-positionals then positionals.

I actually separated it before but refactored it back since it wasn't relevant to the pr. I thought I accidentally left that change in :p